src/AspNet.Security.OAuth.Alipay/AlipayAuthenticationOptions.cs

```diff
@@ -29,5 +29,46 @@ public AlipayAuthenticationOptions()
         ClaimActions.MapJsonKey(Claims.Gender, "gender");
         ClaimActions.MapJsonKey(Claims.Nickname, "nick_name");
         ClaimActions.MapJsonKey(Claims.Province, "province");
+        ClaimActions.MapJsonKey(Claims.OpenId, "open_id");
+        ClaimActions.MapJsonKey(Claims.UserId, "user_id");
+        ClaimActions.MapCustomJson(System.Security.Claims.ClaimTypes.NameIdentifier, user => user.GetString(NameIdentifierKey));
     }
+
+    public override void Validate()
+    {
+        base.Validate();
+
+        if (!string.IsNullOrEmpty(AppCertSN) && !string.IsNullOrEmpty(RootCertSN))
+        {
+            EnableCertSignature = true;
+        }
+        else if (!string.IsNullOrEmpty(AppCertPath) && !string.IsNullOrEmpty(RootCertPath))
+        {
+            try
+            {
+                AppCertSN = AlipayCertificationUtils.GetCertSN(AppCertPath);
+                RootCertSN = AlipayCertificationUtils.GetRootCertSN(RootCertPath);
+                EnableCertSignature = true;
+            }
+            catch (Exception ex)
+            {
+                throw new ArgumentException($"The '{nameof(AppCertPath)}' and '{nameof(RootCertPath)}' options must be set to the correct certificate files.", ex);
+            }
+        }
+    }
+
+    /// <summary>
+    /// See https://opendocs.alipay.com/mini/0ai2i6?pathHash=13dd5946
+    /// </summary>
+    public string NameIdentifierKey { get; set; } = "user_id";
+
+    public string? AppCertPath { get; set; }
+
+    public string? RootCertPath { get; set; }
```

Comment:
Don't assume that certificates are on the file system - they could (should?) be stored in a secure location, such as the Windows Certificate Manager, or loaded from a remote source such as Azure Key Vault. Take a look at the Apple provider for examples on handling such scenarios.